hotel_reservation/models/hotel_reservation.py

```diff
@@ -702,16 +710,76 @@ def get_room_summary(self):
+                            st = ('state', '=', 'assigned')
+                            rline_ids = (reservation_line_obj.search([rl, chi,
+                                                                      cho, st])
+                                         )
+                            for res_room in rline_ids:
+                                rrci = res_room.check_in
+                                rrco = res_room.check_out
+                                cid = datetime.datetime.strptime(rrci, dt)
+                                cod = datetime.datetime.strptime(rrco, dt)
+                                dur = cod - cid
+                                if room_list_stats:
+                                    count = 0
+                                    for rlist in room_list_stats:
+                                        cidst = datetime.datetime.strftime(cid,
+                                                                           dt)
+                                        codst = datetime.datetime.strftime(cod,
+                                                                           dt)
+                                        rm_id = res_room.room_id.id
+                                        ci = rlist.get('date') >= cidst
+                                        co = rlist.get('date') <= codst
+                                        rm = rlist.get('room_id') == rm_id
+                                        st = rlist.get('state') == 'Reserved'
+                                        if ci and co and rm and st:
+                                            count += 1
+                                    if count - dur.days == 0:
+                                        c_id1 = user_obj.browse(self._uid)
+                                        c_id = c_id1.company_id
+                                        con_add = 0
+                                        amin = 0.0
+                                        if c_id:
+                                            con_add = c_id.additional_hours
+                                        if con_add > 0:
+                                            amin = abs(con_add * 60)
+                                            hr_dur = abs((dur.seconds / 60))
+                                        if amin > 0:
+                                            if hr_dur >= amin:
+                                                reservline_ids = True
+                                            else:
+                                                reservline_ids = False
+                                        else:
+                                            if hr_dur > 0:
+                                                reservline_ids = True
+                                            else:
+                                                reservline_ids = False
+                                    else:
+                                        reservline_ids = False
```

Comment:
are you sure we need such a huge code?Write comments each block and explain why whats written